litmus-portal/frontend/src/pages/AnalyticsPage/index.tsx

```diff
@@ -25,17 +78,166 @@ const AnalyticsPage: React.FC = () => {
     { variables: { projectID: selectedProjectID } }
   );
 
-  const workflow = data?.getWorkFlowRuns.filter(
-    (w) => w.workflow_run_id === workflowRunId
-  )[0];
+  const [workflowRunDataForPlot, setWorkflowRunDataForPlot] = React.useState<
+    WorkflowRunData[]
+  >([]);
+
+  useEffect(() => {
+    const workflowRuns: WorkflowRunData[] = [];
+
+    let testDate: string = '';
+
+    const selectedWorkflows = data?.getWorkFlowRuns.filter(
+      (w) => w.workflow_run_id === workflowRunId
+    );
+
+    const workflowRunsExecutionData: ExecutionData[] = [];
+    selectedWorkflows?.forEach((data) => {
+      try {
+        const executionData = JSON.parse(data.execution_data);
+        workflowRunsExecutionData.push(executionData);
+      } catch (error) {
+        console.error(error);
+      }
+    });
+
+    const chaosDataArray: any[] = [];
+    const experimentTestNameArray: string[] = [];
+    const experimentTestVerdictArray: string[] = [];
+    const experimentTestResultsArray: number[] = [];
+
+    workflowRunsExecutionData.forEach((data) => {
+      try {
+        const nodeKeys = Object.keys(data.nodes);
+        const { nodes } = data;
+        const chaosNodeKeys: string[] = [];
+        nodeKeys.forEach((node) => {
+          try {
+            const { name } = data.nodes[node];
+            if (name === 'run-chaos') {
+              chaosNodeKeys.push(node);
+            }
+          } catch (error) {
+            console.error(error);
+          }
+        });
+
+        chaosNodeKeys.forEach((key) => {
+          try {
+            const chaosNode = nodes[key];
+            const chaosNodeData = JSON.parse(JSON.stringify(chaosNode));
```

Comment:
Node is already typed...Json.parse is not required

Reply:
actually the types one doesn't have the chaos data currently.

Reply:
add them to the interface then...